src/Tes.Runner/Transfer/Md5HashListProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace Tes.Runner.Transfer
+{
+    public class Md5HashListProvider : IHashListProvider
+    {
+        private readonly ILogger logger = PipelineLoggerFactory.Create<Md5HashListProvider>();
+        private readonly ConcurrentDictionary<int, string> hashesDictionary = new();
+
+        public ConcurrentDictionary<int, string> HashList => hashesDictionary;
```

Comment:
Consider exposing a read-only snapshot instead of the directory object itself here. This isn't something who's content should be changed outside of this class.

Reply:
No need to do this. This property is only use for testing, and must be public because tests are in another project.